dvc/scm/git/__init__.py

```diff
@@ -324,15 +327,30 @@ def get_tree(self, rev):
         return GitTree(self.repo, self.resolve_rev(rev))
 
     def get_rev(self):
-        return self.repo.git.rev_parse("HEAD")
+        return self.repo.rev_parse("HEAD").hexsha
 
     def resolve_rev(self, rev):
-        from git.exc import GitCommandError
-
+        from git.exc import BadName, GitCommandError
+        from contextlib import suppress
+
+        names = [rev] + ([] if Git.is_sha(rev) else ["origin/" + rev])
```

Comment:
Is it safe to assume `origin` here? This might not always be the case (except for external_repo, I guess).

>

Reply:
On second thought, this is fine because, if the reference does not exist, `BadName` is thrown.

But, we could check if remote exist via `git.repo.remote().exists()` or get a list of remotes  (`git.repo.remotes`) and check for rev.

Reply:
Changed this to more advanced logic to be as close as possible to how `git checkout` behaves.